docs/readmes/orc8r/dev_sub_digests.md

```diff
@@ -0,0 +1,88 @@
+1. Instead of syncing at every interval, only sync when AGW is out-of-sync
+2. Instead of syncing all data, sync the cloud-AGW data changeset when reasonable
+
+## Overview
+
+The architectural additions implemented for the subscriber digests pattern are highlighted below.
+
+![Subscriber Digests Architecture](assets/orc8r/subscriber_digests_architecture.png)
+
+### Digest Generation
+
+This pattern relies on taking deterministically encoded data snapshots, a.k.a. "digests", that are used to represent the local data version. To this end, orc8r library code `mproto` is created using the golang third-party library `proto` to conduct consistently deterministic data serializations.
+
+The subscriber digests pattern specifically generates and utilizes two types of digests
+
+1. **Flat digests**: a digest of the entire set of subscriber data of a network
+2. **Per-subscriber digests**: a digest of a single subscriber data object; for a network, the list of per-subscriber digests is tracked and distributed en masse
+
+NOTE: Currently, a decoupling process between subscriber data objects and their gateway-specific APN resource configurations is conducted to make sure the generated digests is representative and also network-general. See [additional notes](./dev_sub_digests.md#apn-resource-handling) for more details.
+
+### Subscriber Digests Cache
+`subscriberdb_cache` is a single-pod service in sole charge of managing network digests. The service consists mainly of worker code that constantly generates new flat digests and per-subscriber digests at a configurable interval for each network.
+
+The generated digests are written to a cloud SQL store to which `subscriberdb` service has read-only access. In this way, `subscriberdb` can directly read the most recently cached digests from the SQL store, instead of making extraneous gRPC calls over the network and having multiple servicer workers compete for the `subscriberdb_cache` endpoints.
+
+Additionally,  `subscriberdb_cache` also caches the subscriber data that the digests represent. This is to ensure consistency between the cached digests and the actual subscriber data objects in a `subscriberdb` servicer response.
+
+That is, every time an update of the digests are detected, `subscriberdb_cache` loads all subscriber data from `configurator` and applies the batch update to store. In this way, `subscriberdb_cache` acts as the single source of truth from which the `subscriberdb` servicer loads all data required for the subscriber digests pattern.
+
+`subscriberdb_cache` supports loading subscriber data either directly by IDs or by pages. To that end, it generates its own page tokens using the same mechanism as in  `configurator` for simplicity.
+
+### AGW Digest Store
```

Comment:
Nit: h1 headers should be "Title Cased", all other headers should be "Sentence cased"